### PR TITLE
core:fix error load private key in offline model

### PR DESCRIPTION
**Pull request by kjzz**
Fix issue #4978 

License:MIT
Signed-off-by: zhangkejie <601172892@qq.com>

### CODE REVIEW

File: core/core.go [hunk 1]
```diff
@@ -786,9 +786,11 @@ func (n *IpfsNode) SetupOfflineRouting() error {
 		// Routing was already set up
 		return nil
 	}
-	err := n.LoadPrivateKey()
-	if err != nil {
-		return err
+	if n.PrivateKey == nil {
```
[the remark this answers is not in the document]
Comment (kjzz) [reply core/core.go]: i think put it in LoadPrivateKey can not slove this problem,you can see issue #4978,i have fix this issue .what's more u can see the 196 line of the code  in core/command/id.go,we should add the judgment in the project

[the remark this answers is not in the document]
Comment (kjzz) [reply core/core.go]: @Stebalien

**Comment by Kubuxu** [comment]
@kjzz could you add a sharness test for this bugfix. I think the best way to do it is to add it in `test/sharness/t0600-issues-and-regressions-online.sh` start a node with `--offline` parameter (you can do that by passing and argument after `test_launch_ipfs_daemon`) and perform the steps to reproduce after that.

**Comment by kjzz** [comment]
@Kubuxu ok,i will.thank u

**Pull request closed without merging**